pkg/api/util/db_test.go

```diff
@@ -682,3 +682,160 @@ func TestDeleteFileShareAclDBEntry(t *testing.T) {
+		expectedError := "accessTo is empty. Please give valid ip segment"
+		assertTestResult(t, err.Error(), expectedError)
+	})
+
+	t.Run("Invalid Ip Segment", func(t *testing.T) {
+		in.ProfileId, in.Type, in.AccessTo = "d2975ebe-d82c-430f-b28e-f373746a71ca", "ip", "201.100.101.8/9.9"
+		mockClient := new(dbtest.Client)
+		mockClient.On("GetFileShare", context.NewAdminContext(), in.FileShareId).Return(&SampleFileShares[0], nil)
+		mockClient.On("CreateFileShareAcl", context.NewAdminContext(), in).Return(&SampleFileSharesAcl[2], nil)
+		db.C = mockClient
+
+		_, err := CreateFileShareAclDBEntry(context.NewAdminContext(), in)
+		expectedError := fmt.Sprintf("invalid IP segment %v", in.AccessTo)
+		assertTestResult(t, err.Error(), expectedError)
+	})
+
+	t.Run("Invalid Ip", func(t *testing.T) {
+		in.ProfileId, in.Type, in.AccessTo = "d2975ebe-d82c-430f-b28e-f373746a71ca", "ip", "201.100.101"
+		mockClient := new(dbtest.Client)
+		mockClient.On("GetFileShare", context.NewAdminContext(), in.FileShareId).Return(&SampleFileShares[0], nil)
+		mockClient.On("CreateFileShareAcl", context.NewAdminContext(), in).Return(&SampleFileSharesAcl[2], nil)
+		db.C = mockClient
+
+		_, err := CreateFileShareAclDBEntry(context.NewAdminContext(), in)
+		expectedError := fmt.Sprintf("%v is not a valid ip. Please give the proper ip", in.AccessTo)
+		assertTestResult(t, err.Error(), expectedError)
+	})
+
+	t.Run("Empty accesscapability", func(t *testing.T) {
+		in.ProfileId, in.Type, in.AccessTo, in.AccessCapability = "d2975ebe-d82c-430f-b28e-f373746a71ca", "ip", "201.100.101.9", []string{}
+		mockClient := new(dbtest.Client)
+		mockClient.On("GetFileShare", context.NewAdminContext(), in.FileShareId).Return(&SampleFileShares[0], nil)
+		mockClient.On("CreateFileShareAcl", context.NewAdminContext(), in).Return(&SampleFileSharesAcl[2], nil)
+		db.C = mockClient
+
+		_, err := CreateFileShareAclDBEntry(context.NewAdminContext(), in)
+		expectedError := fmt.Sprintf("empty fileshare accesscapability. Supported accesscapability are: {read, write}")
+		assertTestResult(t, err.Error(), expectedError)
+	})
+
+	t.Run("Invalid accesscapabilities", func(t *testing.T) {
+		in.ProfileId, in.Type, in.AccessTo, in.AccessCapability = "d2975ebe-d82c-430f-b28e-f373746a71ca", "ip", "201.100.101.9", []string{"read", "execute"}
+		mockClient := new(dbtest.Client)
+		mockClient.On("GetFileShare", context.NewAdminContext(), in.FileShareId).Return(&SampleFileShares[0], nil)
+		mockClient.On("CreateFileShareAcl", context.NewAdminContext(), in).Return(&SampleFileSharesAcl[2], nil)
+		db.C = mockClient
+
+		value := "execute"
+		_, err := CreateFileShareAclDBEntry(context.NewAdminContext(), in)
+		expectedError := fmt.Sprintf("invalid fileshare accesscapability: %v. Supported accesscapability are: {read, write}", value)
+		assertTestResult(t, err.Error(), expectedError)
+	})
+
+	t.Run("Invalid fileshare id given", func(t *testing.T) {
```

Comment:
AFAIU, the user need NOT give any ID during create file share...the system will create and return an ID, which needs to be used for this fileshare there on.

Reply:
As per current architecture fileshare Id can be provided by user